app/models/workday/commercial_agreements.rb

```diff
@@ -0,0 +1,38 @@
+class Workday::CommercialAgreements
+  def revenue_category_ids
+    result = {}
+    report_entries.each do |report_entry|
+      framework_number = report_entry.at_xpath('wd:Framework_Number').text
+      revenue_category_wid_xml = report_entry.at_xpath('wd:Revenue_Category_ID')
+      result[framework_number] = revenue_category_wid_xml.text if revenue_category_wid_xml
+    end
+    result
```

Comment:
How about using `each_with_object` for this and `#tax_code_ids`?

```ruby
    report_entries.each_with_object({}) do |report_entry, result|
      framework_number = report_entry.at_xpath('wd:Framework_Number').text
      revenue_category_wid_xml = report_entry.at_xpath('wd:Revenue_Category_ID')
      result[framework_number] = revenue_category_wid_xml.text if revenue_category_wid_xml
    end
```

Reply:
That is nicer.